src/Microsoft.AspNet.OData.Shared/Builder/LinkGenerationHelpers.cs

```diff
@@ -550,5 +558,54 @@ internal static IList<ODataPathSegment> GenerateBaseODataPathSegments(this Resou
                 feedContext.EntitySetBase,
                 odataPath);
         }
+
+        private static IList<ODataPathSegment> GenerateContainmentODataPathSegments(this ResourceContext resourceContext)
+        {
+            IList<ODataPathSegment> navigationPathSegments = new List<ODataPathSegment>();
+            ResourceContext currentResourceContext = resourceContext;
+
+            while (currentResourceContext != null)
+            {
+                IEdmNavigationSource navigationSource = currentResourceContext.NavigationSource;
+
+                if (navigationSource is IEdmContainedEntitySet containedEntitySet)
+                {
+                    KeySegment keySegment = new KeySegment(
+                        ConventionsHelpers.GetEntityKey(currentResourceContext),
+                        currentResourceContext.StructuredType as IEdmEntityType,
+                        navigationSource: null);
```

Comment:
@habbes In this scenario where we're generating the path segments for a URL, it'd appear that there's no adverse consequence in leaving `navigationSource` as `null`. I passed a `null` because the same was being done in the existing logic.

Reply:
@habbes On second thoughts, there's no reason to leave it null yet we have the value that we should pass. I have fixed that...